Migrating to AWS SDK V3 | Replace node-http-handler Package

shirady opened the pull request:
### Explain the changes
1. @aws-sdk/node-http-handler was deprecated ([here](https://www.npmjs.com/package/@aws-sdk/node-http-handler)). I found it out through this [aws/aws-sdk-ls-v3#4990](https://github.com/aws/aws-sdk-js-v3/pull/4990).
2. I run `npm uninstall @aws-sdk/node-http-handler` and `npm install @smithy/node-http-handler `.
3. I changed the imports that already used this package from `@aws-sdk/node-http-handler` to `@smithy/node-http-handler`.

### Issues: Fixed #xxx / Gap #xxx
1. none.

### Testing Instructions:
1. none.


- [ ] Doc added/updated
- [ ] Tests added

shirady committed:
@aws-sdk/node-http-handler was deprecated
run npm uninstall @aws-sdk/node-http-handler
run npm install @smithy/node-http-handler

Signed-off-by: shirady <57721533+shirady@users.noreply.github.com>

v-harihar commented:
I am not sure about which is the stable version and what package we need to adopt.

Because `@aws-sdk/node-http-handler` package is marked under `preview` on aws-sdk documentation https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/preview/Package/-aws-sdk-node-http-handler/

`@smithy/node-http-handler` is the latest package available. 
https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-smithy-node-http-handler/

The documentation is not really clear about what is what. 

This [link](https://github.com/aws/aws-sdk-js-v3/discussions/5006) says that `@smithy` is rename to `aws-sdk`. 

May be @dannyzaken @liranmauda can suggest better

shirady commented:
Hi @v-harihar,
I'll try to respond to your comments:

> I am not sure which is the stable version and what package we need to adopt.
> Because `@aws-sdk/node-http-handler` package is marked under `preview` on aws-sdk documentation https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/preview/Package/-aws-sdk-node-http-handler/

The comment about preview is about the documentation API: "This AWS SDK API Reference for JavaScript v3 is in preview and is subject to change." and not about the package itself, the current version is [this](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/index.html). Inside it you can see the package @smithy/node-http-handler ([here](@smithy/node-http-handler)).


> `@smithy/node-http-handler` is the latest package available. https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-smithy-node-http-handler/

I agree, I also added in the description the PR that changed the packages.

> The documentation is not really clear about what is what.
> This [link](https://github.com/aws/aws-sdk-js-v3/discussions/5006) says that `@smithy` is rename to `aws-sdk`.

In the link, it says: "The @aws-sdk/node-http-handler package has become deprecated... Apparently it has now moved to the @smithy". It is fine, we move to use the package under smithy.

@dannyzaken @liranmauda I hope it is clear.
